### PR TITLE
Enabling cross-platform builds and tests.

### PR DESCRIPTION
**Pull request by MikeStall**
Still have some known test failures to address:
https://github.com/microsoft/PowerApps-Language-Tooling/issues/128

### CODE REVIEW

File: src/PAModelTests/UtilityTests.cs [hunk 1]
```diff
@@ -22,7 +23,7 @@ public void TestEscaping(string unescaped, string escaped)
         public void TestNotEscaped()
         {
             // Not escaped.
-            var a = "0123456789AZaz[]_. \\";
```
Comment (MikeStall) [inline src/PAModelTests/UtilityTests.cs:25]: What's the impact of removing / ?

File: targets/Program.cs [hunk 2]
```diff
@@ -131,7 +131,7 @@ static void CleanDirectory(string directoryPath)
                 }
             }
             catch (AccessViolationException) { /* swallow */ }
-            // catch (Exception ex) { Console.WriteLine(ex.Message); }
+            catch (IOException) { Directory.Delete(directoryPath, true); /* try again */ }
```
Comment (MikeStall) [inline targets/Program.cs:134]: What's this?

File: src/PAModel/Utility/Utility.cs [hunk 3]
```diff
@@ -211,7 +211,7 @@ private static bool DontEscapeChar(char ch)
                 (ch == '_') ||
                 (ch == '.') ||
                 (ch == ' ') || // allow spaces, very common.  
-                (ch == '\\'); // Allow directory separators. 
+                (ch == Path.DirectorySeparatorChar); // Allow directory separators.
```
Comment (MikeStall) [inline src/PAModel/Utility/Utility.cs:214]: This doesn't feel right... escaping should be agnostic to OS. 
IE, what if we unpack on windows, check into github, and then pack on Linux?

File: src/PAModel/Checksum/ChecksumMaker.cs [hunk 4]
```diff
@@ -290,7 +291,7 @@ private static void ChecksumJson(Context ctx, IncrementalHash hash, JsonElement
                     else if (ctx.IsXmlDoubleEncoded && !string.IsNullOrWhiteSpace(str))
                     {
                         var parsedXML = XDocument.Parse(str);
-                        var xmlString = parsedXML.ToString(SaveOptions.None);
+                        var xmlString = parsedXML.ToString(SaveOptions.None).Replace("\r\n", "\n");
```
Comment (MikeStall) [inline src/PAModel/Checksum/ChecksumMaker.cs:294]: this looks like the critical spot for checksum? catching the double escaped xml?

Comment (lesaltzm) [reply src/PAModel/Checksum/ChecksumMaker.cs:294]: yup

File: targets/Program.cs (hunk 2 above)
Comment (lesaltzm) [reply targets/Program.cs:134]: whoops, something I was testing out, I was running into an issue where some file handles were sticking around longer than I wanted. not actually needed, I think

File: src/PAModel/Utility/Utility.cs (hunk 3 above)
Comment (lesaltzm) [reply src/PAModel/Utility/Utility.cs:214]: Right, yeah. This works for linux->linux round trip, and windows->windows. I think the solution for linux->windows is we need to escape both directory separators, probably.

**Pull request closed without merging**